rct229/schema/validate.py

```diff
@@ -25,7 +25,36 @@
 
 # def check_schedule_association(rmd)
 
-# def check_fluid_loop_or_piping_association(rmd)
+
+def check_fluid_loop_or_piping_association(rmd):
+    """
+    Check the association between fluid loops or piping and pumps that reference them.
+    Parameters
+    ----------
+    rmd
+
+    Returns list of mismatched fluid loop or piping ids
+    -------
+
+    """
+    mismatch_list = []
+    fluid_loop_or_piping_id_list = find_all("$.fluid_loops[*].id", rmd)
+    fluid_loop_or_piping_id_list.extend(
+        find_all(
+            "$.service_water_heating_distribution_systems[*].service_water_piping[*].id",
+            rmd,
+        )
+    )
+
+    referenced_fluid_loop_or_piping_id_list = find_all(
+        "$.pumps[*].loop_or_piping",
+        rmd,
+    )
+    for fluid_loop_or_piping_id in referenced_fluid_loop_or_piping_id_list:
+        if fluid_loop_or_piping_id not in fluid_loop_or_piping_id_list:
+            mismatch_list.append(fluid_loop_or_piping_id)
+    return mismatch_list
```

Comment:
Similarly, I am not sure if we should allow orphan water heater loops - that sounds OK to me though

Reply:
In this case it would mean there are no pumps in the model which reference the FluidLoop/ServiceWaterPiping. I think there could be cases where a loop may not have a pump so in this case orphans are ok.